### DISCUSSION TITLE
Dockerfile - any interest?

### DISCUSSION THREAD

**Discussion by gilbertgong** [zanllp/infinite-image-browsing#366] [Ideas]
Hi all, I've put together a Dockerfile of running IIB in standalone mode.  Others are welcome to use it or the image built which ends up on docker hub.  I can also do a PR to contribute it to the project if there's interest.  I offered to do so to zanllp and he suggested I initiate a discussion first.

Link to the Dockerfile + associated files:
https://gitlab.com/ggong/docker-utility-images/-/tree/main/sd-webui-infinite-image-browsing?ref_type=heads

Link to the image:
https://hub.docker.com/r/dynafire/sd-webui-infinite-image-browsing

**Comment by ItsNoted** [comment]
I would love to see this updated and actively maintained. It works as is but I think it's running an older version of the app.

**Reply by ianmustafa** [reply]
baeed on the `Dockerfile`, you'll get the latest version of IIB if you're willing to build the image yourself

but yes, the image on Docker Hub was updated 5 months ago

**Reply by du5tball** [reply]
I took some time to rework what @gilbertgong published. Changes:
* The base-image is alpine instead of python:3.10-slim-bullseye
* It uses a virtualenv, as PEP 668 changes the way you can install things with pip (either you add a --break-system-packages flag, or use venvs)
* Rebuilding the image automatically checks out the latest tag.

Link: https://github.com/du5tball/sd-infimage-docker

I don't intend on building an image and uploading it to docker, since it would suffer the same issue as gilbertgong's image: It would not get updated. Additionally, you can auto-build in a docker-compose.yml, so deleting the container and image, then running `docker compose up -d` always rebuilds the container with the most current release tag. I only intend to update my repo with fixes when I rebuild my container and notice the build breaking.

@zanllp You're free to use my files and build a pipeline that automatically runs when you push a new release tag, if you so desire.

**Comment by djnixy** [comment]
I have my own, forked from gilbertgong minus a lot of things I don't need. Note that I may not update it until it breaks.

https://gitlab.com/djnixy/docker-utility-images/-/tree/main?ref_type=heads
nikiakbar/sd-webui-infinite-image-browsing

Builds weekly.

**Comment by PBandDev** [comment]
The GitLab repo linked above is pending deletion. I created an updated Docker setup:

  https://github.com/PBandDev/infinite-image-browser-docker

  Tags:

<table>
  <thead>
  <tr>
  <th>Tag</th>
  <th>Description</th>
  <th>Updates</th>
  </tr>
  </thead>
  <tbody>
  <tr>
  <td><code>latest</code></td>
  <td>Follows upstream <code>main</code></td>
  <td>Weekly</td>
  </tr>
  <tr>
  <td><code>vX.X.X</code></td>
  <td>Pinned releases</td>
  <td>On new release</td>
  </tr>
  </tbody>
  </table>

```
docker run -d -p 8080:8080 \
  -v /path/to/images:/outputs:ro \
  ghcr.io/pbanddev/infinite-image-browser-docker:latest
```
